### PR TITLE
feat(dom): create package

### PR DESCRIPTION
**Pull request by fahrradflucht**

### CODE REVIEW

File: docs/getting-started/introduction.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ provide a full-fledged solution for composing micro frontends:
   React][placing-feature-apps-on-a-web-page-using-react], but also allows the
   integration of micro frontends that are [built with any other frontend
   technology][dom-feature-app] (e.g. Vue.js, Angular, Web Components).
+- [`@feature-hub/dom`][dom-api] – A library for building a Feature Hub
```
Comment (clebert) [inline docs/getting-started/introduction.md]: Maybe we shouldn't advertise the package here yet. The documentation has not yet been adapted and the sentence is rather unfinished.

File: jest.config.js [hunk 2]
```diff
@@ -7,7 +7,9 @@ module.exports = {
     '**/packages/*/src/**/*.{ts,tsx}',
     '!**/packages/demos/src/**/*.{ts,tsx}',
     // Covered by the integration test 'module-loader-amd'
-    '!**/packages/module-loader-amd/src/index.ts'
+    '!**/packages/module-loader-amd/src/index.ts',
+    // Covered by the integration test 'integrator-dom'
+    '!**/packages/dom/**/*'
```
Comment (clebert) [inline jest.config.js]: !**/packages/dom/**/*.{ts,tsx}

File: packages/demos/src/integrator-dom/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    let interceptedPage: Page;
+
+    beforeAll(async () => {
+      interceptedPage = await browser.newPage();
+
+      await interceptedPage.setRequestInterception(true);
+
+      interceptedPage.on('request', async request => {
+        if (!request.url().endsWith('feature-app.umd.js')) {
+          await request.continue();
```
Comment (clebert) [inline packages/demos/src/integrator-dom/index.test.ts]: What effect does this have?

File: packages/demos/src/integrator-dom/integrator.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import {createFeatureHub} from '@feature-hub/core';
+import {createFeatureAppLoader} from '@feature-hub/dom';
+import {loadAmdModule} from '@feature-hub/module-loader-amd';
+
+const {featureAppManager} = createFeatureHub('test:integrator', {
+  moduleLoader: loadAmdModule
+});
+
+createFeatureAppLoader(featureAppManager);
```
Comment (clebert) [inline packages/demos/src/integrator-dom/integrator.ts]: maybe we should rename this method to `defineFeatureAppLoader` or `registerFeatureAppLoader`

File: packages/dom/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# @feature-hub/dom
+
+[![Package Version][package-badge]][package-npm]
+[![Website][website-badge]][website] [![API][api-badge]][api]
+
+A library for building a Feature Hub integrator using Web Components.
```
Comment (clebert) [inline packages/dom/README.md]: Maybe we should link to https://en.wikipedia.org/wiki/Web_Components

File: packages/dom/src/feature-app-container.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+ */
+export interface DomFeatureApp {
+  /**
+   * @param container The container element to which the Feature App can attach
+   * itself.
+   */
+  attachTo(container: Element): void;
+}
+
+export function createFeatureAppContainer(
```
Comment (clebert) [inline packages/dom/src/feature-app-container.ts]: See comment above

File: packages/dom/src/feature-app-container.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  manager: FeatureAppManager
```
Comment (clebert) [inline packages/dom/src/feature-app-container.ts]: I thought we were currently writing the name out everywhere. Isn't that so? `featureAppManager`

File: packages/dom/src/feature-app-container.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+    public disconnectedCallback(): void {
+      if (this.featureAppScope) {
+        this.featureAppScope.destroy();
+      }
+
+      super.disconnectedCallback();
+    }
+
+    private getFeatureApp(): HTMLElement | TemplateResult {
```
Comment (clebert) [inline packages/dom/src/feature-app-container.ts]: I think we can move the entire content of this method to render. I don't like the name and see no reason for outsourcing.

File: packages/dom/src/feature-app-loader.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
+import {FeatureAppManager} from '@feature-hub/core';
+import {LitElement, html, property} from 'lit-element';
+import {TemplateResult} from 'lit-html';
+import {until} from 'lit-html/directives/until';
+import {createFeatureAppContainer} from './feature-app-container';
+
+export function createFeatureAppLoader(
+  manager: FeatureAppManager
```
Comment (clebert) [inline packages/dom/src/feature-app-loader.ts]: See the comments above

File: packages/dom/src/feature-app-loader.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+): typeof HTMLElement {
+  createFeatureAppContainer(manager);
+
+  class FeatureAppLoader extends LitElement {
+    @property({type: String})
+    public src = '';
```
Comment (clebert) [inline packages/dom/src/feature-app-loader.ts]: What is the meaning of an empty string as default value?

File: packages/dom/src/feature-app-loader.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+      `;
+    }
+
+    private async loadFeatureApp(): Promise<TemplateResult> {
+      try {
+        const definition = await manager.getAsyncFeatureAppDefinition(this.src)
+          .promise;
+
+        return html`
+          <feature-app-container
```
Comment (clebert) [inline packages/dom/src/feature-app-loader.ts]: Have you thought about using https://github.com/phaux/babel-jsx-lit-html? I understand that in terms of CI the feature can also come later. Maybe we need an issue for that?

File: wallaby.js [hunk 12]
```diff
@@ -16,7 +16,6 @@ function setup({projectCacheDir, testFramework}) {
 module.exports = wallaby => ({
   files: [
     'jest.config.js',
-    'scripts/setup-test-framework.js',
```
Comment (clebert) [inline wallaby.js]: Don't we need the file anymore?

Comment (KingHenne) [reply wallaby.js]: we already deleted this file previously and forgot about this entry

File: jest.config.js [hunk 13]
```diff
@@ -23,5 +23,6 @@ module.exports = {
   modulePathIgnorePatterns: ['/lib'],
   setupFilesAfterEnv: ['expect-puppeteer'],
   testMatch: ['<rootDir>/packages/*/src/**/*.test.{ts,tsx}'],
-  testURL: 'http://example.com'
+  testURL: 'http://example.com',
+  transformIgnorePatterns: ['node_modules/(?!lit-(html|element))']
```
Comment (KingHenne) [inline jest.config.js]: can be removed

**Pull request merged**